File: QuizAppClient/src/ui/SearchingThread.java
package ui;

import client.ClientMain;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class SearchingThread extends Thread{

    ClientGui frame;
    static boolean isOpponentFound;
    private static boolean isStartPressed;

    public static boolean isStartPressed() {
        try {
            Thread.sleep(30);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        return isStartPressed;
    }

    public static void setStartPressed(boolean startPressed) {
        try {
            Thread.sleep(30);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        isStartPressed = startPressed;
    }


    public boolean isOpponentFound() {
        return isOpponentFound;
    }

    public void setOpponentFound(boolean opponentFound) {
        isOpponentFound = opponentFound;
    }

    public SearchingThread(ClientGui Jframe){
        this.frame = Jframe;
    }

    @Override
    public void run() {
        Font font = new Font(Font.SANS_SERIF, Font.BOLD, 20);

        JTextField nameTxt = new JTextField();
        nameTxt.setBounds(15,30, 280,50);
        nameTxt.setFont(font);
        nameTxt.setBorder(null);
        nameTxt.setEditable(false);

        JButton buttonSubmit = new JButton("Start");
        buttonSubmit.setBounds(90, 210,100,40);
        buttonSubmit.setEnabled(false);

        JDialog dialog = new JDialog(frame,"Name");
        dialog.setLayout(null);
        dialog.setSize(300, 300);
        dialog.add(nameTxt);
        dialog.add(buttonSubmit);
        dialog.setLocationRelativeTo(null);
        dialog.setVisible(true);

        buttonSubmit.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                dialog.dispose();
                SearchingThread.setStartPressed(true);
                //frame.start();
            }
        });
        while (!isOpponentFound()){
            nameTxt.setText("Searching for opponent............");
        }

        nameTxt.setText("Your Opponent = " + ClientMain.opponentName);
        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        buttonSubmit.setEnabled(true);
    }
}

File: QuizAppClient/src/ui/SearchingThread.kt
package ui

import client.ClientMain
import java.awt.Font
import javax.swing.JButton
import javax.swing.JDialog
import javax.swing.JTextField

class SearchingThread(var frame: ClientGui) : Thread() {
    private var startPressed: Boolean = false
    var isOpponentFound: Boolean = false

    override fun run() {
        val font = Font(Font.SANS_SERIF, Font.BOLD, 20)
        val nameTxt = JTextField()
        nameTxt.setBounds(15, 30, 280, 50)
        nameTxt.font = font
        nameTxt.border = null
        nameTxt.isEditable = false
        val buttonSubmit = JButton("Start")
        buttonSubmit.setBounds(90, 210, 100, 40)
        buttonSubmit.isEnabled = false
        val dialog = JDialog(frame, "Name")
        dialog.layout = null
        dialog.setSize(300, 300)
        dialog.add(nameTxt)
        dialog.add(buttonSubmit)
        dialog.setLocationRelativeTo(null)
        dialog.isVisible = true
        buttonSubmit.addActionListener {
            dialog.dispose()
            startPressed = true
            //frame.start();
        }
        while (!isOpponentFound) {
            nameTxt.text = "Searching for opponent............"
        }
        nameTxt.text = "Your Opponent = " + ClientMain.opponentName
        try {
            sleep(1000)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
        buttonSubmit.isEnabled = true
    }

    companion object {
        var isOpponentFound = false
        var isStartPressed = false
            get() {
                try {
                    sleep(30)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
                return field
            }
            set(startPressed) {
                try {
                    sleep(30)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
                field = startPressed
            }

    }

}